src/rt/test/perf/backpressure1/backpressure1.cc

```diff
@@ -0,0 +1,193 @@
+    r.prev = now;
+  }
+};
+
+struct Proxy : public VCown<Proxy>
+{
+  size_t index;
+
+  Proxy(size_t index_) : index(index_) {}
+
+  void trace(ObjectStack& st) const
+  {
+    if (this != proxy_chain.back())
+    {
+      st.push(proxy_chain[index + 1]);
+      return;
+    }
+
+    for (auto* r : receiver_set)
+      st.push(r);
+  }
+};
+
+struct Forward : public VAction<Forward>
+{
+  Proxy* proxy;
+
+  Forward(Proxy* proxy_) : proxy(proxy_) {}
+
+  void f()
+  {
+    if (proxy != proxy_chain.back())
+    {
+      auto* next = proxy_chain[proxy->index + 1];
+      Cown::schedule<Forward>(next, next);
+      return;
+    }
+
+    Cown::schedule<Receive>(receiver_set.size(), (Cown**)receiver_set.data());
+  }
+};
+
+struct Sender : public VCown<Sender>
+{
+  using clk = std::chrono::steady_clock;
+
+  clk::time_point start = clk::now();
+  std::chrono::milliseconds duration;
+
+  Sender(std::chrono::milliseconds duration_) : duration(duration_) {}
+
+  void trace(ObjectStack& st) const
+  {
+    if (proxy_chain.size() > 0)
+    {
+      st.push(proxy_chain[0]);
+      return;
+    }
+
+    for (auto* r : receiver_set)
+      st.push(r);
+  }
+};
+
+struct Send : public VAction<Send>
+{
+  Sender* s;
+
+  Send(Sender* s_) : s(s_) {}
+
+  void f()
+  {
+    if (proxy_chain.size() > 0)
+      Cown::schedule<Forward>(proxy_chain[0], proxy_chain[0]);
+    else
+      Cown::schedule<Receive>(receiver_set.size(), (Cown**)receiver_set.data());
+
+    if ((Sender::clk::now() - s->start) < s->duration)
+      Cown::schedule<Send>(s, s);
+  }
+};
+
+int main(int argc, char** argv)
+{
+  opt::Opt opt(argc, argv);
+  auto seed = opt.is<size_t>("--seed", 5489);
+  auto cores = opt.is<size_t>("--cores", 4);
+  auto senders = opt.is<size_t>("--senders", 100);
+  auto receivers = opt.is<size_t>("--receivers", 1);
+  auto proxies = opt.is<size_t>("--proxies", 0);
+  auto duration = opt.is<size_t>("--duration", 10'000);
+  logger::cout() << "cores: " << cores << ", senders: " << senders
+                 << ", receivers: " << receivers << ", duration: " << duration
+                 << "ms" << std::endl;
+
+  auto& sched = Scheduler::get();
+  Scheduler::set_detect_leaks(true);
+#ifdef USE_SYSTEMATIC_TESTING
+  Systematic::enable_logging();
+  sched.set_seed(seed);
+#else
+  UNUSED(seed);
+#endif
+  sched.set_fair(true);
+  sched.init(cores);
+
+  auto* alloc = ThreadAlloc::get();
+
+  for (size_t r = 0; r < receivers; r++)
+    receiver_set.push_back(new (alloc) Receiver);
+
+  for (size_t p = 0; p < proxies; p++)
+    proxy_chain.push_back(new (alloc) Proxy(p));
+
+  Scheduler::set_allow_teardown(false);
+  auto thr = std::thread([=] {
+    for (size_t i = 0; i < senders; i++)
+    {
+      if (proxy_chain.size() > 0)
+      {
+        Cown::acquire(proxy_chain[0]);
+      }
+      else
+      {
+        for (auto* r : receiver_set)
+          Cown::acquire(r);
+      }
+
+      auto* s = new (alloc) Sender(std::chrono::milliseconds(duration));
```

Comment:
This could be +- a random percentage to make the test more robust.

Reply:
Systematic testing would be a better option if we wanted to cover possible interleavings at the end. Existing `func-` prefixed tests cover that with the `coin()`s inserted to randomly trigger backpressure conditions.

Reply:
makes sense